easybuild/tools/module_generator.py

```diff
@@ -257,10 +246,41 @@ def prepend_paths(self, key, paths, allow_abs=False, expand_relpaths=True):
         :param allow_abs: allow providing of absolute paths
         :param expand_relpaths: expand relative paths into absolute paths (by prefixing install dir)
         """
+        return self.update_paths(key, paths, prepend=True, allow_abs=allow_abs, expand_relpaths=expand_relpaths)
+
+    def update_paths(self, key, paths, prepend=True, allow_abs=False, expand_relpaths=True):
+        """
+        Generate append/prepend-path statements for the given list of paths.
+
+        :param key: environment variable to append paths to
+        :param paths: list of paths to append
+        :param allow_abs: allow providing of absolute paths
+        :param expand_relpaths: expand relative paths into absolute paths (by prefixing install dir)
+        """
+        if isinstance(paths, str):
+            self.log.debug("Wrapping %s into a list before using it for %s", paths, key)
+            paths = [paths]
+
         paths = self._filter_paths(key, paths)
```

Comment:
I think this should be done in `_update_paths`, as I can imagine that an EC (somehow ends up) setting both PYTHONPATH and EBPYTHONPREFIXES and with this logic here you could end up setting the latter 2 times depending on the order as you unconditionally add EBPYTHONPREFIXES in https://github.com/easybuilders/easybuild-framework/pull/4496/files#diff-4af72e9777d353325a29df400ab4229e14defd653e413db51786dc2172d3baefR278